modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/db/wal/WalDeletionArchiveAbstractTest.java

```diff
@@ -309,6 +314,95 @@ public void testSingleCleanWalArchive() throws Exception {
         assertEquals(logStrs.toString(), 1, logStrs.size());
     }
 
+    /**
+     * Check that if the maximum archive size is equal to one WAL segment size, then when the archive is overflowing,
+     * segments that are needed for binary recovery will not be deleted from the archive until the checkpoint occurs
+     * and finishes.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testMaxWalArchiveSizeEqualsOneWalSegmentSize() throws Exception {
+        int walSegmentSize = (int)MB;
+
+        IgniteConfiguration cfg = getConfiguration(getTestIgniteInstanceName(0))
+            .setCacheConfiguration(cacheConfiguration())
+            .setDataStorageConfiguration(
+                new DataStorageConfiguration()
+                    .setCheckpointFrequency(Long.MAX_VALUE)
+                    .setMaxWalArchiveSize(walSegmentSize)
+                    .setWalSegmentSize(walSegmentSize)
+                    .setWalSegments(2)
+                    .setDefaultDataRegionConfiguration(
+                        new DataRegionConfiguration()
+                            .setPersistenceEnabled(true)
+                            .setMaxSize(GB)
+                            .setCheckpointPageBufferSize(GB)
+                    )
+            );
+
+        IgniteEx n = startGrid(cfg);
+
+        n.cluster().state(ClusterState.ACTIVE);
+
+        // Let's not let a checkpoint happen.
+        gridDatabase(n).checkpointReadLock();
+
+        try {
+            // Let's reserve the very first segment that will definitely be needed for the checkpoint.
+            assertTrue(wal(n).reserve(new FileWALPointer(0, 0, 0)));
```

Comment:
maybe we can extract `wal(n)` to a variable?

Reply:
Fix it